pkg/log/context.go

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package log
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/sirupsen/logrus"
+)
+
+type logKey struct{}
+
+// LevelKey is the context key for the log level
+type LevelKey struct{}
+
+var (
+	supportedFormatters = map[string]logrus.Formatter{
+		"json": &logrus.JSONFormatter{},
+		"text": &logrus.TextFormatter{},
+	}
+	mux          = sync.Mutex{}
+	once         = sync.Once{}
+	defaultEntry = logrus.NewEntry(logrus.StandardLogger())
+	// C is an alias for ForContext
+	C = ForContext
+	// R is an alias for ForContextProvider
+	R = ForContextProvider
+	// D is an alias for Default
+	D = Default
+)
+
+const (
+	// FieldComponentName is the key of the component field in the log message.
+	FieldComponentName = "component"
+	// FieldCorrelationID is the key of the correlation id field in the log message.
+	FieldCorrelationID = "correlation_id"
```

Comment:
do we want to set correlation id for outgoing requests too?  this might help troubleshooting errors / logs between us and the underlying actual brokers

Reply:
It is set in https://github.com/Peripli/service-manager/blob/logging/pkg/util/client.go#L65